nixos/modules/services/network-filesystems/juicefs.nix

```diff
@@ -0,0 +1,200 @@
+
+      extraArgs = lib.mkOption {
+        type = lib.types.listOf lib.types.str;
+        default = [ ];
+        description = "Additional arguments for mount command";
+        example = [
+          "--no-verify-ssl"
+          "--cache-partial-only"
+        ];
+      };
+    };
+  };
+
+in
+{
+  options.services.juicefs = {
+    enable = lib.mkEnableOption "JuiceFS Service";
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.juicefs;
+      description = "JuiceFS package to use";
+    };
+
+    volumes = lib.mkOption {
+      type = lib.types.attrsOf volumeOptions;
+      default = { };
+      description = "Set of JuiceFS volumes to mount";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    # Create cache directories
+    systemd.tmpfiles.rules = lib.concatLists (
+      lib.mapAttrsToList (name: volume: [
+        "d ${volume.cacheDir} 0755 root root -"
+      ]) cfg.volumes
+    );
+
+    systemd.services = lib.mapAttrs' (
+      name: volume:
+      lib.nameValuePair "juicefs-${name}" {
+        description = "JuiceFS volume ${name}";
+
+        wantedBy = [ "multi-user.target" ];
+        wants = [ "network-online.target" ];
+        requires = [ "network-online.target" ];
+        after = [ "network-online.target" ];
+
+        serviceConfig = {
+          Type = "forking";
+
+          # Environment variables for credentials and settings
+          Environment =
+            [
+              "JFS_NO_UPDATE_FSTAB=1"
+              "AWS_ENDPOINT=${volume.storageOptions.endpoint or ""}"
+            ]
+            ++ lib.optional (volume.accessKey != "") "ACCESS_KEY=${volume.accessKey}"
+            ++ lib.optional (volume.secretKey != "") "SECRET_KEY=${volume.secretKey}";
+
+          # Initialize the volume if needed
+          ExecStartPre = pkgs.writeShellScript "juicefs-init-${name}" ''
+            set -e
+            export PATH="${pkgs.coreutils}/bin:$PATH"
+
+            # Check if volume needs formatting
+            if ! ${cfg.package}/bin/juicefs status ${volume.metaUrl} >/dev/null 2>&1; then
+              echo "Formatting JuiceFS volume ${volume.name}..."
+              ${cfg.package}/bin/juicefs format \
+                ${volume.metaUrl} \
+                ${volume.name} \
+                --storage ${volume.storage} \
+                --bucket ${volume.bucket} \
+                --compress ${volume.compress}
+            else
+              echo "JuiceFS volume ${volume.name} already formatted."
+            fi
+
+            # Create required directories
+            mkdir -p ${volume.mountPoint}
+            mkdir -p ${volume.cacheDir}
+          '';
+
+          # Mount the volume
+          ExecStart = pkgs.writeShellScript "juicefs-mount-${name}" ''
+            set -e
+            ${cfg.package}/bin/juicefs mount \
+              ${volume.metaUrl} \
+              ${volume.mountPoint} \
+              --background \
```

Comment:
Can't we just let systemd handle service without it needing to daemonize? This would proably require dropping this line and the `Type = "forking";` above.